packages/vite/src/index.ts

```diff
@@ -0,0 +1,45 @@
+import type { Plugin as VitePlugin } from 'vite';
+import type { UserConfig } from '@vocab/core';
+
+import { transformVocabFile } from './transform-vocab-file';
+import { virtualResourceLoader } from './virtual-resource-loader';
+
+import { trace } from './logger';
+
+import { compiledVocabFileFilter, virtualModuleId } from './consts';
+
+export type VocabPluginOptions = {
+  vocabConfig: UserConfig;
+};
+
+export default function vitePluginVocab({
+  vocabConfig,
+}: VocabPluginOptions): VitePlugin {
+  trace(
+    `Creating Vocab plugin${
+      vocabConfig ? ` with config file ${vocabConfig}` : ''
+    }`,
+  );
+  return {
+    name: 'vite-plugin-vocab',
+    resolveId(id) {
+      if (id.includes(virtualModuleId)) {
+        return id;
+      }
+    },
+    load(id) {
+      if (id.includes(virtualModuleId)) {
+        return virtualResourceLoader(id);
+      }
```

Comment:
The other part of the [virtual modules convention](https://vite.dev/guide/api-plugin.html#virtual-modules-convention) involves prefixing the id with `\0`. Unsure if this is _really_ worth doing doing, but worth chatting about.

Reply:
Yeah I can have a deeper look into this and see how the virtual modules get created and resolved within vite.